bin/specextract

```diff
@@ -798,28 +652,113 @@ def group_by_obsid(file_stack, file_type):
 ##########################################################################
 
 def mk_asphist(asol_param, evt_filename_filter, full_outroot, dtffile, instrument, chip_id, verbose, clobber, tmpdir):
+    """
+    return per chip aspect histogram and file name string
+    """
 
-    #asp_out = "{0}_asphist{1}.fits".format(full_outroot,chip_id) #full_outroot+"_asphist"+str(event_stats(infile, "ccd_id"))+".fits"
     asp_out = tempfile.NamedTemporaryFile(suffix="_asphist{}".format(chip_id),dir=tmpdir)
 
     asphist.punlearn()
 
-    asphist.infile  = asol_param  # single file, @files.lis, or
-                                  # comma-separated list of files
+    asphist.infile = asol_param # single file, @files.lis, or
+                                # comma-separated list of files
 
     if instrument == "ACIS":
-        asphist.evtfile = "{0}[ccd_id={1}]".format(evt_filename_filter,chip_id) #evt_filename+"[ccd_id="+str(event_stats(infile, "ccd_id"))+"]"
+        asphist.evtfile = "{0}[ccd_id={1}]".format(evt_filename_filter,chip_id)
         asphist.dtffile = ""
     else:
-        asphist.evtfile = "{0}[chip_id={1}]".format(evt_filename_filter,chip_id) #evt_filename+"[chip_id="+str(event_stats(infile, "chip_id"))+"]"
+        asphist.evtfile = "{0}[chip_id={1}]".format(evt_filename_filter,chip_id)
         asphist.dtffile = dtffile
 
     asphist.outfile = asp_out.name
     asphist.verbose = verbose
     asphist.clobber = "yes"
 
-    return asphist(), asp_out.name, asp_out
+    asphist()
+    
+    return asp_out.name, asp_out
+
+
+def clip_wmap(wmapfile,threshold,tmpdir):
+    """
+    OPTIONAL: Truncate sky2tdet WMAP to speed up mkwarf by threshold 
+              clipping the WMAP for large areas. Rebinning the WMAP 
+              is less desirable since it essentially can randomly cause 
+              badpixels/columns/etc to be over or under weighted.  (It 
+              will be the same every time, just random in that you don't 
+              have control over it.)  If your region is huge (i.e. whole 
+              chip) then that probably won't matter, if it's a modest size, 
+              eg off-axis point-like source, then binning is bad.
+    
+              One way to threshold (and there are several) is just to look 
+              at the pixel values in the WMAP, and determine a cutoff that 
+              preserves a certain amount of the total flux. That way, if a 
+              badpixel/column/etc. is not weighted by a lot of flux it can 
+              be explicitly purged and likewise if they are covered by a 
+              large flux, then they will be preserved and the ARF 
+              appropriately weighted.
+    """
+
+    cr = pcr.read_file("{}[wmap]".format(wmapfile))
+    im = cr.get_image().values
+    threshold = float(threshold)
+
+    cr.__del__()
```

Comment:
I'm not sure what deleting the crate does to the values - I would suggest either

    im = cr.get_image().values.copy()

or just remove the `cr.__del__()` call (I prefer the second option).